.github/workflows/application.yml

```diff
@@ -0,0 +1,47 @@
+name: Application
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+
+concurrency:
+  group: application-pipeline-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    uses: ./.github/workflows/test.yml
+
+  build-image:
+    if: github.event_name == 'pull_request'
+    strategy:
+      matrix:
+        env: [dev, poc]
+    name: Build docker image for "${{ matrix.env }}" environment
+    needs: test
+    uses: ./.github/workflows/build-and-push.yml
+    with:
+      push: false
+      environment: ${{ matrix.env }}
+    secrets:
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_ACCOUNT: ${{ secrets.AWS_ACCOUNT }}
+
+  build-and-push-image:
+    if: github.event_name == 'push' && github.ref_name == 'main'
+    strategy:
+      matrix:
+        env: [dev, poc]
+    name: Build and push docker image for "${{ matrix.env }}" environment to ECR
+    needs: test
+    uses: ./.github/workflows/build-and-push.yml
+    with:
+      push: true
```

Comment:
Is it simpler to calculate the boolean here from the if block above, and then just have a "build-and-push" that may or may not actually push, at this level?

Reply:
Done